src/contexts/ElderlyContext.tsx

```diff
@@ -4,30 +4,38 @@ import { getElderlyProfile } from '../utils/elderly/profile';
 import { ElderlyHomeProfile, Module } from './../dto/modules/modules.dto';
 import useAsyncEffect from './../hooks/useAsyncEffect';
 import { getElderlyCode } from '../utils/elderly/code';
+import { getCaretakerList } from '../utils/elderly/caretakerList';
+import { User } from '../dto/modules/user.dto';
+import { ElderlyLinkCode } from '../dto/modules/code.dto';
 
 export interface ElderlyContextProps {
   elderlyProfile: ElderlyHomeProfile | undefined;
   setElderlyProfile: (val: ElderlyHomeProfile) => void;
   moduleList: Module[];
   setModuleList: (val: Module[]) => void;
-  elderlyCode: string;
-  setElderlyCode: (val: string) => void;
+  elderlyCode: ElderlyLinkCode;
+  setElderlyCode: (val: ElderlyLinkCode) => void;
+  caretakerList: User[];
+  setCaretakerList: (val: User[]) => void;
 }
 
 export const ElderlyContext = createContext({} as ElderlyContextProps);
 
 const ElderlyContextProvider = ({ ...props }) => {
   const [elderlyProfile, setElderlyProfile] = useState<ElderlyHomeProfile>();
   const [moduleList, setModuleList] = useState<Module[]>([]); // available modules and its name
-  const [elderlyCode, setElderlyCode] = useState<string>('');
+  const [elderlyCode, setElderlyCode] = useState<ElderlyLinkCode>({"code": ''});
```

Comment:
could you save state as string? it's much easier to understand. Use setState(response.code).